From 7090583eab63d85004e2547362cef6140bd6d486 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Fri, 6 May 2022 10:59:05 -0700
Subject: [PATCH 2/2] fakelinux: Don't let CFLAGS leak in from environment

On Arch Linux, makepkg.conf defines a few CFLAGS, which are exported
globally during package building:

CFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions \
        -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security \
        -fstack-clash-protection -fcf-protection"

This causes issues when running the tests for Hexagon, as fakelinux's
Makefile appends its compiler flags to CFLAGS, rather than overriding
them:

=================================== FAILURES ===================================
________________________ TestArchitecture.test_hexagon _________________________

self = <test.test_build.TestArchitecture object at 0x7fe2ea1ab7c0>
linux = local('/tmp/pytest-of-builduser/pytest-0/source0/linux')

    @pytest.mark.skipif(shutil.which("ld.lld") is None, reason="requires lld")
    @pytest.mark.skipif(clang_version < 10, reason="requires clang 10+")
    def test_hexagon(self, linux):
        result = build(tree=linux, target_arch="hexagon", toolchain="clang")
>       assert result.passed
E       assert False
E        +  where False = <tuxmake.build.Build object at 0x7fe2ea013c70>.passed

test/test_build.py:468: AssertionError
----------------------------- Captured stdout call -----------------------------
...
clang-13: warning: argument unused during compilation: '-fstack-clash-protection' [-Wunused-command-line-argument]
error: option 'cf-protection=return' cannot be specified on this target
error: option 'cf-protection=branch' cannot be specified on this target
2 errors generated.
...

Define CFLAGS as a simple variable (":=") so that CFLAGS does not leak
in from the environment.

Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 test/fakelinux/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/fakelinux/Makefile b/test/fakelinux/Makefile
index 8a874b3..fda642e 100644
--- a/test/fakelinux/Makefile
+++ b/test/fakelinux/Makefile
@@ -14,7 +14,7 @@ CC     = clang
 LD     = ld.lld
 endif
 
-CFLAGS += -nostdlib
+CFLAGS := -nostdlib
 ifneq ($(CROSS_COMPILE),)
 ifneq ($(shell $(CC) --version 2>&1 | head -n1 | grep clang),)
 triplet := $(patsubst %-,%,$(CROSS_COMPILE))
-- 
2.36.0

